### PR TITLE
fix: Select disabled state hover style, Combobox disabled state open on chevron click

### PR DESCRIPTION
**Pull request by smhigley**
This PR includes the following fixes for disabled states for Combobox, Dropdown, and Select:

- Select does not show hover/active styles when disabled
- Adds stories for Combobox/Dropdown disabled state
- Combobox does not open on chevron click when disabled
- Dropdown button text color when disabled

Part of #26069

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Avatar-mount_base.html">1553</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Avatar-mount.html">1559</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Button-mount_base.html">1096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Button-mount.html">1098</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProvider-mount_base.html">1848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProvider-mount.html">1842</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-mount_base.html">725</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-mount.html">732</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">687</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">690</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">719</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">714</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/MakeStyles-mount_base.html">2207</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/MakeStyles-mount.html">2192</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Persona-mount_base.html">3444</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/Persona-mount.html">3460</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/SpinButton-mount_base.html">2872</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26068/perf-test-react-components/SpinButton-mount.html">2865</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `78.629 kB`<br />`25.167 kB` | `78.643 kB`<br />`25.176 kB` |   `14 B` <br />`9 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                    | `75.831 kB`<br />`25.116 kB` | `75.845 kB`<br />`25.123 kB` |   `14 B` <br />`7 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `77.876 kB`<br />`25.104 kB` | `78.043 kB`<br />`25.131 kB` | `167 B` <br />`27 B`  |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                    |  `23.474 kB`<br />`8.017 kB` |  `23.535 kB`<br />`8.039 kB` |  `61 B` <br />`22 B`  |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>                          | `31.731 kB`<br />`10.256 kB` | `31.792 kB`<br />`10.277 kB` |  `61 B` <br />`21 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `59.381 kB`<br />`16.465 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.067 kB`<br />`52.466 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.843 kB`<br />`11.129 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/70cc602fa95cb0bcb615f2a5cd0ece9992a98494'>70cc602fa95cb0bcb615f2a5cd0ece9992a98494</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-select-f9db8e0b-3088-4a68-9179-d5821aebf0cd.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: no interactive hover style when disabled\"",
```
Comment (khmakoto) [inline change/@fluentui-react-select-f9db8e0b-3088-4a68-9179-d5821aebf0cd.json:3]:
```suggestion
  "comment": "fix: no interactive hover style when disabled",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**